### PR TITLE
replaced busymachines media categoriser with correct one

### PR DESCRIPTION
**Pull request by Luthien-in-edhil**
Sorry for the large diff, I also did some cleaning up

### CODE REVIEW

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
 import org.apache.solr.client.solrj.response.FacetField;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.http.MediaType;
+//import org.springframework.http.MediaType;
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Please delete commented out code

File: api2-war/src/main/java/eu/europeana/api2/v2/utils/ModelUtils.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
     private static final String SOLRFACETS      = "solrfacets";
     private static final String CUSTOMFACETS    = "customfacets";
     private static final String DEFAULT         = "DEFAULT";
+//    private static final String NEWLINE         = System.lineSeparator();
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/utils/ModelUtils.java]: Please delete commented out code

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 3]
```diff
@@ -89,15 +93,22 @@
...
+    private static final String PORTAL                    = "portal";
+    private static final String FACETS                    = "facets";
+    private static final String DEBUG                     = "debug";
+    private static final String SPELLING                  = "spelling";
+    private static final String BREADCRUMB                = "breadcrumb";
+    private static final String FACET_RANGE               = "facet.range";
+    private static final String HITS                      = "hits";
     private static final String ERROR_RETRIEVE_ATTRIBUTES = "error retrieving attributes";
+    private static final String TITLE                     = "title";
+    private static final String _TITLE                    = "_title";
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Please delete unused variable

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 4]
```diff
@@ -89,15 +93,22 @@
...
+    private static final String DESCRIPTION               = "description";
+    private static final String _DESCRIPTION              = "_description";
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Please delete unused variable

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 5]
```diff
@@ -390,53 +418,60 @@ public ModelAndView searchJsonGet(
...
+                                         Boolean media,
+                                         Boolean thumbnail,
+                                         Boolean fullText,
+                                         Boolean landingPage,
+                                         List<Integer> filterTags) {
+        boolean      hasImageRefinements = false;
+        boolean      hasAudioRefinements = false;
+        boolean      hasVideoRefinements = false;
+        boolean      hasTextRefinements  = false;
+        boolean      hasOtherRefinements = false;
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: This variable is unused, is that correct? Or should it be used somewhere? If not, please delete

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 6]
```diff
@@ -390,53 +418,60 @@ public ModelAndView searchJsonGet(
...
+        FacetEncoder facetEncoder        = new FacetEncoder();
+        List<String> newRefinements      = new ArrayList<>();
+
+        Set<MimeTypeEncoding>   imageMimeTypeRefinements      = new HashSet<>();
+        Set<MimeTypeEncoding>   audioMimeTypeRefinements      = new HashSet<>();
+        Set<MimeTypeEncoding>   videoMimeTypeRefinements      = new HashSet<>();
+        Set<MimeTypeEncoding>   textMimeTypeRefinements       = new HashSet<>();
+        Set<MimeTypeEncoding>   otherMimeTypeRefinements      = new HashSet<>();
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: This variable is unused, is that correct? Or should it be used somewhere? If not, please delete

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 7]
```diff
@@ -127,43 +138,46 @@
     @Value("${api.search.hl.MaxAnalyzedChars}")
     private String hlMaxAnalyzedChars;
 
+    public static void main(String args[]) {
+        String[] profiles = {"amit", "rahul", "surya"};
+        System.out.println(StringUtils.join(profiles, ","));
+    }
+
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Please delete testing code, or change it into a proper unit test

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 8]
```diff
@@ -127,43 +138,46 @@
...
-            "/api/v2/search.json",
-            "/record/v2/search.json",
-            "/record/search.json"
-    }, produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ModelAndView searchJsonPost(
-            @RequestParam(value = "wskey") String apikey,
-            @RequestBody SearchRequest searchRequest,
-            HttpServletRequest request,
-            HttpServletResponse response) throws EuropeanaException {
+    @PostMapping(value = {"/api/v2/search.json", "/v2/search.json", "/record/v2/search.json", "/record/search.json"},
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: "/v2/search.json" is not a proper endpoint we want to advertise

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 9]
```diff
@@ -173,33 +187,31 @@ public ModelAndView searchJsonPost(
...
-            @RequestParam(value = "media", required = false) Boolean media,
-            @RequestParam(value = "text_fulltext", required = false) Boolean fullText,
-            @RequestParam(value = "landingpage", required = false) Boolean landingPage,
-            @RequestParam(value = "cursor", required = false) String cursorMark,
-            @RequestParam(value = "callback", required = false) String callback,
-            @SolrEscape @RequestParam(value = "hit.fl", required = false) String hlFl,
-            @RequestParam(value = "hit.selectors", required = false) String hlSelectors,
-            HttpServletRequest request,
-            HttpServletResponse response) throws EuropeanaException {
+    @GetMapping(value = {"/api/v2/search.json", "/v2/search.json", "/record/v2/search.json", "/record/search.json"},
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: "/v2/search.json" is not a endpoint we want to advertise

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 10]
```diff
@@ -294,43 +300,63 @@ public ModelAndView searchJsonGet(
...
+                                                                                                                   "1")
+                                                                                                           .setParameter(
+                                                                                                                   "fl",
+                                                                                                                   IdBeanImpl
+                                                                                                                           .getFields(
+                                                                                                                                   getBeanImpl(
+                                                                                                                                           clazz)))
+                                                                                                           .setSpellcheckAllowed(
+                                                                                                                   false);
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Please change the indentation of this code so it's better readable

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 11]
```diff
@@ -907,49 +954,60 @@ public ModelAndView fieldTripRss(
...
+                                                      .get(DESCRIPTION)) : gftChannelAttributes.get(
+                        reqLanguage + "_" + DESCRIPTION);
+                channel.language = gftChannelAttributes.get(LANGUAGE) == null ||
+                                   gftChannelAttributes.get(LANGUAGE).equalsIgnoreCase("") ? "--" : gftChannelAttributes
+                                           .get(LANGUAGE);
+                channel.link = gftChannelAttributes.get(LINK) == null || gftChannelAttributes.get(LINK)
+                                                                                             .equalsIgnoreCase("") ? "no link defined" : gftChannelAttributes
+                                       .get(LINK);
+                channel.image = gftChannelAttributes.get(IMAGE) == null ||
+                                gftChannelAttributes.get(IMAGE).equalsIgnoreCase("") ? null : new FieldTripImage(
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Indentation was bad before, but I feel the altered indentation only made things worse

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java [hunk 12]
```diff
@@ -969,23 +1027,21 @@ public ModelAndView fieldTripRss(
...
+     * Method to find all encoded facets in tags
      *
      * @return the JSON response
      */
     @SwaggerIgnore
-    @GetMapping(value = "/api/v2/tagdecoder.json", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ModelAndView searchJson(
-            @RequestParam(value = "tag") String tag) {
+    @GetMapping(value = {"/v2/decodetags.json", "/v2/tagdecoder.json"},
+                produces = org.springframework.http.MediaType.APPLICATION_JSON_VALUE)
```
Comment (P-Ehlert) [inline api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Endpoints should start with /api (for consistency)

**Review by P-Ehlert — CHANGES_REQUESTED** [review]

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 7 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: I don't know where that method comes from: it suddenly appears on dec 15 on my local history overview, but I certainly have not consciously added it. I don't even understand what the point of it is.

Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: I'll take it out.

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 9 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: fixed

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 8 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: fixed

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 10 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: done

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 11 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Agreed. I tried some other formatting settings for binary and ternary operations, I think it looks somewhat better now.

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 6 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Indeed! That one was an artefact of the previous unclarity re. textual tech facets. It's no longer needed and I removed it.

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 5 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: deleted

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 4 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: deleted

File: api2-war/src/main/java/eu/europeana/api2/v2/utils/ModelUtils.java (hunk 2 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/utils/ModelUtils.java]: done

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 3 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: done

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 1 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: done

File: api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java (hunk 12 above)
Comment (Luthien-in-edhil) [reply api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java]: Done.

**Pull request merged**